.github/scripts/capture-pre-upgrade.tsx

```
@@ -0,0 +1,66 @@
+import { readFileSync, writeFileSync, existsSync } from "fs";
+
+interface PackageJson {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  [key: string]: any;
+}
+
+interface PreUpgradeState {
+  dependencies: Record<string, string>;
+  devDependencies: Record<string, string>;
+  timestamp: string;
+}
+
+function validatePackageJson(): void {
+  if (!existsSync("package.json")) {
+    console.error("❌ package.json not found in current directory");
+    process.exit(1);
+  }
+}
+
+function readPackageJson(): PackageJson {
+  try {
+    return JSON.parse(readFileSync("package.json", "utf8"));
+  } catch (error) {
+    console.error("❌ Error reading package.json:", error);
+    process.exit(1);
+  }
+}
+
+function savePreUpgradeState(state: PreUpgradeState): void {
+  try {
+    writeFileSync("pre-upgrade-state.json", JSON.stringify(state, null, 2));
+    writeFileSync("package.json.before", readFileSync("package.json", "utf8"));
+  } catch (error) {
+    console.error("❌ Error saving pre-upgrade state:", error);
+    process.exit(1);
+  }
+}
+
+async function capturePreUpgrade(): Promise<void> {
+  try {
+    console.log("📦 Capturing pre-upgrade state...");
+
+    validatePackageJson();
+    const packageJson = readPackageJson();
+
+    const state: PreUpgradeState = {
+      dependencies: packageJson.dependencies || {},
+      devDependencies: packageJson.devDependencies || {},
+      timestamp: new Date().toISOString(),
+    };
+
+    savePreUpgradeState(state);
+
+    const totalDeps =
+      Object.keys(state.dependencies).length +
+      Object.keys(state.devDependencies).length;
+    console.log(`✅ Pre-upgrade state captured (${totalDeps} dependencies)`);
+  } catch (error) {
+    console.error("❌ Error capturing pre-upgrade state:", error);
+    process.exit(1);
+  }
+}
+
+capturePreUpgrade();
```

so generally it is much simpler if can get rid of all the explicit try/catches in these scripts and just let the error propagate to the top:

```suggestion
capturePreUpgrade().catch(e => console.error(e));
```

This will give stack traces and make it a lot easier to debug the problem, rather than adding process.exit(1) all over the place and then trying to match error strings to code.  (this is true for all the scripts)

---

Can you actually remove all the try/catchs from the scripts, whenever all they're doing is calling process.exit()? And add .catch() to each of the scripts as well